### PR TITLE
Create binary_search.py

### LINKED ISSUE #377: BinarySearch in Python2
This is an open issue for binary search algorithm implementation in Python2. If you wish to contribute, just make some noise and we will assign you.

### PR DESCRIPTION
**Pull request by avsd47**
**Fixes #377 **

By submitting this pull request I confirm I've read and complied with the below declarations.

-   ✔ I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   ✔ I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   ✔My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   ✔This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   ✔This pull request shall only be merged once the build passes.

-   ✔I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    else: 
+      return mid
+  
+  return -1
+  
+def main():
+  arr = [1,2,3,4,5,6]
+  index = binarysearch(arr,5)
+  if (index==-1):
+    print "The element is not found!"
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:25]: Codacy found an issue: [invalid syntax (F999)](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277963)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+def binarysearch(arr,x):
+  low=0
+  high=len(arr)-1
+  mid=0
+
+  while (low<=high):
+
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:7]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277963)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    
+    mid=(low+high)//2
+    
+    if (arr[mid]<x):
+      low=mid+1
+    
+    elif (arr[mid]>x):
+      high=mid-1
+    
+    else:
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:16]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277963)

**Comment by RavinduSachintha** [comment]
this is a duplicated PR. please just use a single PR for a particular issue.

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 2 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:7]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277963)

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 3 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:16]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277963)

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 1 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:25]: Codacy found an issue: [invalid syntax (F999)](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277963)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+def main():
+  arr = [1,2,3,4,5,6]
+  index = binarysearch(arr,5)
+  if(index==-1):
+    print "The element is not found!"
+  else:
+    print "Element is found at index:",index
+if __name__ == '__main__':
+    main()
+
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:23]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277963)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  low=0
+  high=len(arr)-1
+  mid=0
+  while(low<=high):
+    mid=(low+high)//2
+    if (arr[mid]<x):
+      low=mid+1
+    elif (arr[mid]>x):
+      high=mid-1
+    else:
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:11]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277963)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    elif (arr[mid]>x):
+      high=mid-1
+    else: 
+      return mid
+  return -1
+def main():
+  arr = [1,2,3,4,5,6]
+  index = binarysearch(arr,5)
+  if(index==-1):
+    print "The element is not found!"
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:18]: Codacy found an issue: [invalid syntax (F999)](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277963)

**Comment by avsd47** [comment]
Please review it again let me know again of there is an issue this is my first time handling github so please bear me i am learning.

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+def main():
+  arr = [1,2,3,4,5,6]
+  index = binarysearch(arr,5)
+  if(index==-1):
+    print 'The element is not found!'
+  else:
+    print 'Element is found at index:',index
+if __name__ == '__main__':
+    main()
+
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:23]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277963)

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 5 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:11]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277963)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    elif (arr[mid]>x):
+      high=mid-1
+    else: 
+      return mid
+  return -1
+def main():
+  arr = [1,2,3,4,5,6]
+  index = binarysearch(arr,5)
+  if(index==-1):
+    print 'The element is not found!'
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:18]: Codacy found an issue: [invalid syntax (F999)](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277963)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  low=0
+  high=len(arr)-1
+  mid=0
+  while(low<=high):
+    mid=(low+high)//2
+    if(arr[mid]<x):
+      low=mid+1
+    elif(arr[mid]>x):
+      high=mid-1
+    else:
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:11]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277963)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    elif(arr[mid]>x):
+      high=mid-1
+    else: 
+      return mid
+  return -1
+def main():
+  arr = [1,2,3,4,5,6]
+  index = binarysearch(arr,5)
+  if(index==-1):
+    print "The element is not found!"
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:18]: Codacy found an issue: [invalid syntax (F999)](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277963)

**Pull request closed without merging**